addons/website_crm/controllers/main.py

```diff
@@ -60,6 +60,21 @@ def _handle_website_form(self, model_name, **kwargs):
     def insert_record(self, request, model, values, custom, meta=None):
         is_lead_model = model.model == 'crm.lead'
         if is_lead_model:
+            values_email_normalized = tools.email_normalize(values.get('email_from'))
+            if values_email_normalized:
+                visitor = request.env['website.visitor']._get_visitor_from_request()
+                visitor_partner = visitor.partner_id
+                if visitor and visitor_partner and visitor_partner.email_normalized == values_email_normalized:
+                    # Here, 'phone' in values has already been formatted, see _handle_website_form.
+                    values_phone = values.get('phone')
+                    # We write partner id on crm only if no phone exists on partner or in input,
+                    # or if both (after formating) are the same. This way we get additional phone
+                    # if possible, without modifying an existing one.
+                    if values_phone and visitor_partner.phone:
+                        if visitor_partner._phone_format(visitor_partner.phone) == values_phone:
+                            values['partner_id'] = visitor_partner.id
+                    else:
+                        values['partner_id'] = visitor_partner.id
```

Comment:
pas testé mais en lisant le code...

Donc si visitor.phone is False je ne passe pas dans le IF
donc je passe dans le else qui set lead.partner_id avec visitor.partner_id
Du coup tu perds le phone du formulaire a priori, vu que phone deviendra le related de partner_id.phone qui etait a False

non?

Reply:
Hello, merci pour le retour. 
C'est un peu tricky, mais sur le model crm lead il y a une fonction inverse qui écrit sur le partner le phone si il n'y en a pas, ou si il est changé directement sur le form du lead.
Donc je viens de revérifier, si il y a un phone dans le form et pas sur le partner, il est bien présent sur le lead, et écrit aussi sur partner qui n'en avait pas.
Donc à priori ça marche :)